settings: Make page scrollable

MritunjayTiwari14 opened the pull request:
| Before | After (Screenshot) |
| ------- | ------------------ |
| ![column_l](https://github.com/user-attachments/assets/5218af2f-3fe8-445a-8380-69694b0f7301) | ![listview_l](https://github.com/user-attachments/assets/21f84210-aeba-49cb-859d-1c7389e6bb84) |
| ![column_d](https://github.com/user-attachments/assets/d4722bf7-5ecd-419f-9013-2799a3022111) | ![listview_d](https://github.com/user-attachments/assets/d949e2bb-199e-470c-8361-a99187f3848b) |


<br>

### Video Recordings

### Before

https://github.com/user-attachments/assets/03716367-b5d9-4f83-84f4-3716abb59829

<br>

### After (Normal)

https://github.com/user-attachments/assets/92a05eab-6789-4811-bb95-271838c864f3

<br>

### After (Increased Text Size to unreasonably high in android settings to test edge case)

https://github.com/user-attachments/assets/c070507f-8a4f-4fcf-8f4f-5731bef5f9d3


Fixes #1904

chrisbobbe commented on test/widgets/settings_test.dart:
```diff
@@ -160,6 +160,24 @@ void main() {
     }, variant: TargetPlatformVariant({TargetPlatform.android, TargetPlatform.iOS}));
   });
 
+  group('SettingPageScrollBehavior', () {
```
This string looks _like_ an identifier, like the name of a class or something, but then I don't find anything when I look for its definition. To be less confusing, we should just say "SettingsPage scroll behavior" or similar. (`SettingsPage` is the name of a class, so it doesn't add confusion to use that name.)

chrisbobbe commented on test/widgets/settings_test.dart:
```diff
@@ -160,6 +160,24 @@ void main() {
+
+      await tester.scrollUntilVisible(lastElementFinder, 100,
+        scrollable: find.byType(Scrollable));
+      check(lastElementFinder.evaluate().isEmpty).equals(false);
```
nit: how about `check(lastElementFinder).findsOne();

chrisbobbe commented on test/widgets/settings_test.dart:
```diff
@@ -160,6 +160,24 @@ void main() {
   });
 
+  group('SettingPageScrollBehavior', () {
+    testWidgets('scroll settings list when screen size is small', (tester) async {
```
nit:

```suggestion
    testWidgets('content is scrollable when taller than a screenful', (tester) async {
```

(Helps focus the reader on what the test's purpose is, and acknowledges that the test can still be helpful when the screen is normal-sized, in a future where the list of settings is much longer.)

chrisbobbe reviewed (commented):
Thanks! Comments below.

For the commit message:

```
settings: Add scroll for settings page

This commit replaced Column with
ListView Widget to enable scrolling effect.
```

- It needs a `Fixes #<issue number>.` line, like other commits that fix issues in the tracker.
- The sentence in the body isn't needed; it just restates what's already obvious from reading the diff.
- For the summary line, "add scroll" sounds a little odd to me; how about `settings: Make page scrollable`.

MritunjayTiwari14 commented:
Thank you @chrisbobbe for taking time to review, all the requested changes and some other nits have been pushed.

chrisbobbe commented:
Bump on https://github.com/zulip/zulip-flutter/pull/1911#pullrequestreview-3373037117 :
> * It needs a `Fixes #<issue number>.` line, like other commits that fix issues in the tracker.

Bump on https://github.com/zulip/zulip-flutter/pull/1911#discussion_r2457609946 :
> […] we should just say "SettingsPage scroll behavior" or similar. (`SettingsPage` is the name of a class, so it doesn't add confusion to use that name.)

chrisbobbe reviewed (changes requested):
Thanks!

From the Git history (see [Greg's "secret" to using `git log -p`](https://github.com/zulip/zulip-mobile/blob/main/docs/howto/git.md#2-the-secret-to-using-git-log--p), you can see we tend to use
1. "Fixes #1904." at the top, or
2. "Fixes: #1904" at the bottom

but not "Fixes #1904". (I think we're converging on option 1, which is what the Flutter project uses?)

chrisbobbe commented:
Marking for Greg's review; thank you.

gnprice commented on test/widgets/settings_test.dart:
```diff
@@ -64,6 +64,24 @@ void main() {
   }
 
+  group('SettingsPage scroll behavior', () {
+    testWidgets('content is scrollable when taller than a screenful', (tester) async {
```
nit: can simplify by not wrapping this test in a group (since it was the only thing there anyway):
```suggestion
  testWidgets('SettingsPage content is scrollable when taller than a screenful', (tester) async {
```

gnprice replied:
bump — I do think the "SettingsPage" part is helpful, for setting context when this test case goes by in `flutter test` output

gnprice reviewed (commented):
Thanks @MritunjayTiwari14 for taking care of this, and thanks @chrisbobbe for the previous reviews! Just small comments below.

MritunjayTiwari14 commented:
Thank you @gnprice for the review. The following has been changed in the latest commit:

- Unwrapped the test `content is scrollable when taller than a screenful` out of group `SettingsPage scroll behavior` and removed the group.

- Added a new line between the summary and body so that `Fixes #1904.` appears at the top of body not along with summary.

- Removed `scrollable` from `tester.scrollUntilVisible` since flutter automatically finds the Scrollable by default for `scrollable`.

MritunjayTiwari14 commented:
Yes @gnprice, it would be right to provide context instead of just saying content, even though the test belong to SettingsPage . The context of content within SettingsPage can vary. So in the latest changes the following has been accommodated:

- Changed `content` to `SettingsPage` to provide a better context of test.

gnprice commented:
Thanks! Looks good; merging.
